src/test/java/com/microsoft/sqlserver/jdbc/unit/statement/BatchExecutionTest.java

```diff
@@ -167,6 +230,61 @@ public void testExecuteBatch1() {
             fail(TestResource.getResource("R_executeBatchFailed") + ": " + e.getMessage());
         }
     }
+    
+    public void testExecuteBatch1UseBulkCopyAPI() {
+        int i = 0;
+        int retValue[] = {0, 0, 0};
+        int updCountLength = 0;
+        try {
+            String sPrepStmt = "update ctstable2 set PRICE=PRICE*20 where TYPE_ID=?";
+
+            pstmt = connection.prepareStatement(sPrepStmt);
+            Field f1 = SQLServerConnection.class.getDeclaredField("isAzureDW");
+            f1.setAccessible(true);
+            f1.set(connection, true);
+            pstmt.setInt(1, 1);
+            pstmt.addBatch();
+
+            pstmt.setInt(1, 2);
+            pstmt.addBatch();
+
+            pstmt.setInt(1, 3);
+            pstmt.addBatch();
+
+            int[] updateCount = pstmt.executeBatch();
+            updCountLength = updateCount.length;
+
+            assertTrue(updCountLength == 3, "executeBatch does not execute the Batch of SQL statements, Call to executeBatch is Failed!");
+
+            String sPrepStmt1 = "select count(*) from ctstable2 where TYPE_ID=?";
+
+            pstmt1 = connection.prepareStatement(sPrepStmt1);
+
+            for (int n = 1; n <= 3; n++) {
+                pstmt1.setInt(1, n);
+                rs = pstmt1.executeQuery();
+                rs.next();
+                retValue[i++] = rs.getInt(1);
+            }
+
+            pstmt1.close();
```

Comment:
Better move this into a try-with-resources block

Reply:
It's handled at the end of the test anyhoo.

Reply:
The connection itself is available in the class, they're all static objects (don't know why), but why create a new one anywhere else in the class?

Reply:
the modifyConnectionForBulkCopyAPI method modifies the connection object. If we don't create a new connection object (with the try block), the tests that come after this one will be affected by the change I made to the connection (since the only other connection object available is static). Therefore, to prevent that happening, I need to make a new connection object just to test my PR.

Also, they connection object is static because originally the test could just re-use the same connection object that doesn't need to change, but now we need to change our connection object accordingly.

Reply:
> It's handled at the end of the test anyhoo.

 - Not if `executeQuery` fails.

Reply:
i think it'll be closed in the terminateVariation() (this is what I meant by end of the test)